### ISSUE TITLE
Odd "Error: The script will never generate a response." error with esbuild and top-level await

### ISSUE THREAD

**Issue by IgorMinar** [cloudflare/workerd#402]
I've tried to use esbuild with workers, but rain into odd issue with esbuild and top-level await in workerd.

I don't really know what's going on here but it seems that something is off with our task/microtask scheduling from the top-level context (outside of a request context).

I created a repro with instructions in README.md: https://github.com/IgorMinar/repro-workers-sdk-2029-top-level-await

This repro contains several patches and workarounds for other issues that I identified and reduced while trying to get esbuild to work. The repro also contains a workaround for this last scheduling issue, please see the readme.

It would be awesome if someone familiar with the workerd could take a look at the repo and root-cause this problem or reduce the issue further. Thanks

**Comment by IgorMinar** [comment]
btw I have a hunch that something about esbuild's use of WASM and input and output streams to manage comms between the JS context and WASM is causing the pending work not to be detected.

Maybe we focus solely on monitoring the status of the microtask queue and not take WASM into account when we throw the "The script will never generate a response" error?

**Comment by kentonv** [comment]
Wasm doesn't have its own task queue AFAIK.

Given that this apparently reproduces even with just one request, I think the usual problems with sharing promises between requests couldn't be the issue here. It seems like esbuild must legitimately be waiting on a promise that never resolves, for some reason. Perhaps the code which was supposed to resolve the promise threw an exception before doing so?
